src/com/google/javascript/jscomp/VarCheck.java

```diff
@@ -303,7 +307,12 @@ public void visit(NodeTraversal t, Node n, Node parent) {
             }
             // fall through
           default:
-            t.report(n, NAME_REFERENCE_IN_EXTERNS_ERROR, n.getString());
+            // Don't warn for simple var assignments "/** @const */ var foo = bar;"
+            // They are used to infer the types of namespace aliases.
+            if (parent.getType() != Token.NAME || parent.getParent() == null ||
+                !allowedExternAssignmentTypes.contains(parent.getParent().getType())) {
```

Comment:
Sorry I should have said this before. We have NodeUtil.isNameDeclaration which does exactly this.